Amethyst/Layout/Layout.swift

```diff
@@ -115,15 +131,40 @@ protocol Layout {
     var windowActivityCache: WindowActivityCache { get }
 
     func reflow(_ windows: [SIWindow], on screen: NSScreen) -> ReflowOperation
+    func assignedFrame(_ window: SIWindow, of windows: [SIWindow], on screen: NSScreen) -> FrameAssignment?
 }
 
 protocol PanedLayout {
+    var mainPaneRatio: CGFloat { get }
+    func setMainPaneRawRatio(rawRatio: CGFloat)
```

Comment:
`shrinkMainPane()` and `expandMainPane()` were already in the protocol, even though their behavior for some layouts is a no-op. So it seemed natural to expose the boilerplate `mainPaneRatio` that was being used to support that feature, even though some layouts ignore it.

There are many ways to do this, and I don't have a preference.  I'm just following the existing implementation, which was in terms of ratios.

Reply:
The shrink and expand actions are abstract operations. The ratio is an implementation detail.

Reply:
I'm happy to do it another way if you have something in mind, but (from my reading of the code) the concept of a ratio between a "main" pane and the others seems central to the idea of a `PanedLayout`.  How else would you specify the spatial relationship here?

Reply:
For context, the problem being solved here is that the `kAXWindowResizedNotification` listener must, without knowing any specifics about the layout, trigger a reflow that will put the window edge exactly where the cursor left it (using nothing more than the resized window's frame information).

The way I'm currently tackling this is to expose `FrameAssignment` and `ResizeRules` for the windows in a layout, which allows me to work backwards from the resized window's frame  and determine what layout parameters would have produced that size.   Then I expose the setting of those parameters via the layout (in this case, the ratio).

Reply:
Okay. I think I'm convinced on it being a property of paned layouts in general. It's kind of nit-picking, but I think I would prefer having a method that suggests a ratio rather than just a kind of setter. Maybe
```swift
func reccomendMainPaneRatio(_ ratio: CGFloat)
```
It feels a bit more flexible to me as the layout still maintains the logic of how to actually apply the ratio change. That's just a setter for most (all current?) cases, but could be more complex in the future.

Reply:
Sounds good